### ISSUE TITLE
HTML widgets in R Notebooks do not survive switch into visual mode

### ISSUE THREAD

**Issue by jmcphers** [rstudio/rstudio#8227]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.984

### Steps to reproduce the problem

1. Ensure that RStudio is set to create new R Markdown documents in source (not visual) mode
2. Create a new R Notebook (File -> New -> R Notebook)
3. Add a code chunk with the following contents:

```r
library(dygraphs)
lungDeaths <- cbind(mdeaths, fdeaths)
dygraph(lungDeaths)
```

4. Save the notebook with any file name (e.g. `foo.Rmd`)
5. Run the code chunk; an interactive graphic appears
6. Switch the editor into visual mode

### Describe the problem in detail

The interactive graphic does not load in visual mode. Instead, an error is shown.

![image](https://user-images.githubusercontent.com/470418/97617280-cc889480-19da-11eb-8ff5-b8d004f6b996.png)

### Describe the behavior you expected

The interactive graphic should appear in visual mode.

**Comment by jmcphers** [comment]
This problem appears to arise from the fact that R Notebooks have separated "saved" and "unsaved" notebook caches (which is necessary so that any changes you make can discarded if you choose not to save). Switching into visual mode triggers a save as a side effect, which moves HTML widgets from the unsaved cache to the saved cache. The URL of the iframe, however, still refers to the unsaved cache.

**Comment by kfeinauer** [comment]
Verified fixed 1.4.1027
